internal/v5/resources.go

```diff
@@ -189,3 +218,29 @@ func fromResourceDoc(doc *mongodoc.Resource, resources map[string]resource.Meta)
 	r.Revision = doc.Revision
 	return r, nil
 }
+
+func parseResourceId(path string) (mongodoc.ResourceRevision, error) {
+	i := strings.Index(path, "/")
+	if i == -1 {
+		return mongodoc.ResourceRevision{
+			Name:     path,
+			Revision: -1,
+		}, nil
+	}
+	revno, err := strconv.Atoi(path[i+1:])
+	if err != nil {
+		return mongodoc.ResourceRevision{}, errgo.Newf("malformed revision number")
+	}
+	if revno < 0 {
+		return mongodoc.ResourceRevision{}, errgo.Newf("negative revision number")
+	}
+	return mongodoc.ResourceRevision{
+		Name:     path[0:i],
+		Revision: revno,
+	}, nil
+}
+
+func validResourceName(name string) bool {
+	// TODO we should probably be more restrictive than this.
```

Comment:
Let's leave it for now, as the set of allowed chars is eminently suitable for bikeshedding...